web/olga/analytics/tests/test_views.py

```diff
@@ -86,43 +88,89 @@ class TestAccessTokenRegistration(TestCase):
     Tests for access token registration.
     """
 
-    def test_create_edx_installation(self):
+    def test_create_new_edx_instance(self):
         """
-        Verify that create_edx_installation method create one EdxInstallation object.
+        Verify that new installation created after call of create_new_edx_instance call.
         """
+        uid = get_random_string()
         access_token = uuid.uuid4().hex
-        AccessTokenRegistration().create_edx_installation(access_token)
+        AccessTokenRegistration().create_new_edx_instance(access_token, uid)
 
         self.assertEqual(1, EdxInstallation.objects.all().count())
 
+    def test_create_new_edx_instance_double_with_same_uid(self):
+        """
+        Verify that only one new installation create after double call of create_new_edx_instance call with same uid.
+        """
+        uid = get_random_string()
+        access_token, is_need_create = AccessTokenRegistration().get_access_token(uid)
+        self.assertEqual(is_need_create, True)
+        AccessTokenRegistration().create_new_edx_instance(access_token, uid)
+        access_token, is_need_create = AccessTokenRegistration().get_access_token(uid)
+
+        self.assertEqual(is_need_create, False)
+        self.assertEqual(1, EdxInstallation.objects.all().count())
+
+    def test_create_new_edx_instance_double_with_different_uid(self):
+        """
+        Verify that two new installations are created after double create_new_edx_instance call with different uid.
+        """
+        uid = get_random_string()
+        access_token = uuid.uuid4().hex
+        AccessTokenRegistration().create_new_edx_instance(access_token, uid)
+
+        uid = get_random_string()
```

Comment:
I would change name for the second uid to the second_uid or use `get_random_string()` function inside `create_new_edx_instance()`